### PR TITLE
Update Java Package Discovery

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Update Logic for Java Package discovery

### CODE REVIEW

File: packages/python-packages/doc-warden/warden/warden_common.py [hunk 1]
```diff
@@ -184,29 +183,13 @@ def check_folder_against_exclusion_list(folder, path_exclusion_list):
...
 # given a pom.xml, crack it open and ensure that it is actually a package pom (versus a parent pom)
 def is_java_pom_package_pom(file_path):
     root = parse_pom(file_path)
-    jar_tag = root.find('packaging')
+    artifactIdTag = root.find('parent/artifactId')
 
-    if jar_tag is not None:
-        return jar_tag.text == 'jar'
+    if artifactIdTag is not None:
+        return artifactIdTag.text == 'azure-client-sdk-parent' or artifactIdTag.text == 'azure-data-sdk-parent'
```
Comment (scbedd) [inline packages/python-packages/doc-warden/warden/warden_common.py]: Changing away from `all poms with jar packaging` to `any pom that has a parent artifact id`. Am I missing anything?

Comment (chidozieononiwu) [reply packages/python-packages/doc-warden/warden/warden_common.py]: Well all the ones with artifactId set to `azure-client-sdk-parent` or `azure-data-sdk-parent`

Comment (scbedd) [reply packages/python-packages/doc-warden/warden/warden_common.py]: Fair. Alrighty.

**Pull request merged**